Data_Package_Documentation/IGSN_Formatting/recreate_SESAR_template.R

```diff
@@ -0,0 +1,369 @@
+
+# Assumptions: 
+  # Requires input IGSN file to have 2 columns: `IGSN` that has the sample IGSNs and `Parent_IGSN` that has the site IGSNs.
+  # The input file IGSNs should not have the full DOI URL. It is okay if they look like "10.58052/IEPRS007S" or "IEPRS007S".
+  # If there are more than 100 samples or sites, you will have to repeat the downloading process (but this script will walk you through that).
+
+# Directions: 
+  # Go through this script line by line for directions on how to navigate
+  # downloading info from the SESAR database and creating the pseudo template.
+
+# Status: 
+  # complete. 
+  # script has not yet been tested when there are more than 100 sites or samples
+
+
+### Prep Script ################################################################
+
+# load libraries
+library(tidyverse)
+library(devtools)
+library(rlog)
+library(clipr)
+library(readxl)
+library(testthat)
+
+# load functions
+source_url("https://raw.githubusercontent.com/river-corridors-sfa/rcsfa-data_processing_for_publication/refs/heads/main/Data_Transformation/functions/rename_column_headers.R")
+
+# user inputs - update these each time you run the script
+data_package_igsns <- read_csv("Z:/00_ESSDIVE/01_Study_DPs/00_ARCHIVE-WHEN-PUBLISHED/SFA_SpatialStudy_2021_SampleData_v3/v3_SFA_SpatialStudy_2021_SampleData/SPS_Sample_IGSN-Mapping.csv", skip = 1) # read in the file from your data package that has IGSN and Parent_IGSN cols
+ess_dive_infrastructure_only_dir <- "Z:/00_ESSDIVE/01_Study_DPs/00_ARCHIVE-WHEN-PUBLISHED/SFA_SpatialStudy_2021_SampleData_v3/ESS_DIVE_Infrastructure_ONLY"
+
+# create and open your temp directory - this folder and files in it will automatically delete when your R session ends
+temp_dir <- tempdir()
+shell.exec(temp_dir)
+
+# function to download from SESAR
+download_from_SESAR <- function(type = c("sites", "samples"), max) {
+  
+  # go to website
+  browseURL("https://www.geosamples.org/search-options/catalog-search")
+  
+  cat("Your IGSNs have been copied to your clipboard.")
+  cat("\n")
+  cat("In the 'search by multiple IGSNs' box, paste in your site IGSNs. ")
+  cat("\n")
+  cat("Scroll down and click the blue 'search' button.")
+  
+  readline(prompt = "Ready for the next step? After you clicked 'search', enter 'Y' into the console: ")
+  
+  # copies temp directory to your clipboard
+  write_clip(paste0(temp_dir, "/", type))
+  
+  cat('Click the grey "Download" button.')
+  cat("\n")
+  cat("When the dialog box to save the file comes up, paste the temp_dir URL (it's been saved to your clipboard) into the folder path. No need to rename the file, click save.")
+  cat("\n")
+  
+  # check the number of files in the sites dir
+  while(list.files(paste0(temp_dir, "/", type)) %>% length() != i) {
+    
+    cat("If you did not get this dialog box, your file was likely saved to your default download directory. If this happens, manually move the file from your Downloads to the temp directory.")
```

Comment:
Need to specify if its going into the site or samples subfolder

Reply:
To clarify, you want the comment to say something like "manually move the file to the temp directory/Samples"?

Reply:
Comment now looks like this: 
![image](https://github.com/user-attachments/assets/91a04b33-732d-4d13-828b-cac13faaab63)